src/internal/event.ts

```diff
@@ -10,41 +10,90 @@
 
 import { AptosConfig } from "../api/aptos_config";
 import { AccountAddress } from "../core";
-import { AnyNumber, GetEventsResponse, HexInput } from "../types";
+import { AnyNumber, GetEventsResponse, HexInput, IndexerPaginationArgs, MoveResourceType, OrderBy } from "../types";
 import { GetEventsQuery } from "../types/generated/operations";
 import { GetEvents } from "../types/generated/queries";
+import { EventsBoolExp } from "../types/generated/types";
 import { queryIndexer } from "./general";
 
-/**
- * Get events by creation number and the address
- *
- * @param args.aptosConfig - The aptos config
- * @param args.address - The account address
- * @param args.creationNumber - The event creation number
- * @returns Promise<GetEventsByCreationNumberResponse>
- */
-export async function getEventsByCreationNumber(args: {
+export async function getAccountEventsByCreationNumber(args: {
   aptosConfig: AptosConfig;
   address: HexInput;
   creationNumber: AnyNumber;
+  options?: {
```

Comment:
does it make sense to paginate this?  

"Event types are globally identifiable by an account address and monotonically increasing creation_number, one per event type emitted to the given account. This API returns events corresponding to that that event type."

Reply:
check out this [query](https://cloud.hasura.io/public/graphiql?endpoint=https%3A%2F%2Findexer-devnet.staging.gcp.aptosdev.com%2Fv1%2Fgraphql&query=query+MyQuery+%7B%0A++events%28where%3A+%7Bcreation_number%3A+%7B_eq%3A+%223%22%7D%7D%29+%7B%0A++++type%0A++++transaction_version%0A++++transaction_block_height%0A++++sequence_number%0A++++event_index%0A++++data%0A++++creation_number%0A++++account_address%0A++%7D%0A%7D%0A)

Reply:
sorry this query was without account address - I guess you are right

Reply:
done